crates/cli/src/package.rs

```diff
@@ -52,7 +52,7 @@ async fn internal_fetch(
     // TODO: skip when it already exists in store?
     let cas_paths = download_tarball_to_store(
         tarball_cache,
-        &config.store_dir,
+        Arc::new(config.store_dir.to_path_buf()),
```

Comment:
Can you make `config.store_dir` also `Arc<Path>`?

Since `config.store_dir.to_path_buf()` is still an expensive clone that is done one per each `internal_fetch` (i.e. same as before), only when `Npmrc::store_dir` (or `config.store_dir`) already is an `Arc<Path>` would this PR have any meaning.